api/client/service/update.go

```diff
@@ -357,7 +357,12 @@ func removeItems(
 func updateMounts(flags *pflag.FlagSet, mounts *[]mounttypes.Mount) {
 	if flags.Changed(flagMountAdd) {
 		values := flags.Lookup(flagMountAdd).Value.(*MountOpt).Value()
-		*mounts = append(*mounts, values...)
+
+		exist := findAndReplaceMountData(mounts, values)
```

Comment:
So, mount ordering needs to be consistent for the update. It might be easier to build a target key, add all the mounts, error on duplicate. Then, build the mount set and sort it.